go/nomdl/parser_test.go

```diff
@@ -0,0 +1,157 @@
+		suite.assertParseType("Blob"+string(r), types.BlobType)
+	}
+}
+
+func (suite *ParserSuite) TestComments() {
+	suite.assertParseType("/* */Blob", types.BlobType)
+	suite.assertParseType("Blob/* */", types.BlobType)
+	suite.assertParseType("Blob//", types.BlobType)
+	suite.assertParseType("//\nBlob", types.BlobType)
+}
+
+func (suite *ParserSuite) TestCompoundTypes() {
+	suite.assertParseType("List<>", types.MakeListType(types.MakeUnionType()))
+	suite.assertParseType("List<Bool>", types.MakeListType(types.BoolType))
+	suite.assertParseError("List<Bool, Number>", `Unexpected token ",", expected ">", example:1:11`)
+
+	suite.assertParseType("Set<>", types.MakeSetType(types.MakeUnionType()))
+	suite.assertParseType("Set<Bool>", types.MakeSetType(types.BoolType))
+	suite.assertParseError("Set<Bool, Number>", `Unexpected token ",", expected ">", example:1:10`)
+
+	suite.assertParseError("Ref<>", `Unexpected token ">", example:1:6`)
+	suite.assertParseType("Ref<Bool>", types.MakeRefType(types.BoolType))
+	suite.assertParseError("Ref<Number, Bool>", `Unexpected token ",", expected ">", example:1:12`)
+
+	suite.assertParseType("Cycle<42>", types.MakeCycleType(42))
+	suite.assertParseError("Cycle<-123>", `Unexpected token "-", expected Int, example:1:8`)
+	suite.assertParseError("Cycle<12.3>", `Unexpected token Float, expected Int, example:1:11`)
+
+	suite.assertParseType("Map<>", types.MakeMapType(types.MakeUnionType(), types.MakeUnionType()))
+	suite.assertParseType("Map<Bool, String>", types.MakeMapType(types.BoolType, types.StringType))
+	suite.assertParseError("Map<Bool,>", `Unexpected token ">", example:1:11`)
+	suite.assertParseError("Map<,Bool>", `Unexpected token ",", example:1:6`)
+	suite.assertParseError("Map<,>", `Unexpected token ",", example:1:6`)
+}
+
+func (suite *ParserSuite) TestStructTypes() {
+	suite.assertParseType("struct {}", types.MakeStructTypeFromFields("", types.FieldMap{}))
+	suite.assertParseType("struct S {}", types.MakeStructTypeFromFields("S", types.FieldMap{}))
+
+	suite.assertParseType(`struct S {
+                x: Number
+                }`, types.MakeStructTypeFromFields("S", types.FieldMap{
+		"x": types.NumberType,
+	}))
+
+	suite.assertParseType(`struct S {
+                x: Number,
+        }`, types.MakeStructTypeFromFields("S", types.FieldMap{
+		"x": types.NumberType,
+	}))
+
+	suite.assertParseType(`struct S {
+                x: Number,
+                y: String
+        }`, types.MakeStructTypeFromFields("S", types.FieldMap{
+		"x": types.NumberType,
+		"y": types.StringType,
+	}))
+
+	suite.assertParseType(`struct S {
+                x: Number,
+                y: String,
+        }`, types.MakeStructTypeFromFields("S", types.FieldMap{
+		"x": types.NumberType,
+		"y": types.StringType,
+	}))
+
+	suite.assertParseError(`struct S {
+                x: Number
+                y: String
+        }`, `Unexpected token Ident, expected ",", example:3:18`)
+}
```

Comment:
I think that, from the code and the spec above, nested struct types are supported, right?

```
struct S {
  t: struct { n: Number }
}
```

It'd be good to have a test case for complicated stuff like that, and the same with structs definitions nested inside collection definitions and stuff